core/shared/src/main/scala/scalaz/zio/stream/Sink.scala

```diff
@@ -621,14 +634,14 @@ object Sink {
         type State = (Side[E, self.State, B], Side[E1, that.State, (Chunk[A], C)])
 
         val l: ZIO[R, Nothing, Step[Either[E, self.State], Nothing]]  = self.initial.either.map(sequence)
-        val r: ZIO[R, Nothing, Step[Either[E1, that.State], Nothing]] = that.initial.either.map(sequence)
+        val r: ZIO[R1, Nothing, Step[Either[E1, that.State], Nothing]] = that.initial.either.map(sequence)
 
-        val initial: ZIO[R, E1, Step[State, Nothing]] =
+        val initial: ZIO[R1, E1, Step[State, Nothing]] =
           l.zipWithPar(r) { (l, r) =>
             Step.both(Step.leftMap(l)(eitherToSide), Step.leftMap(r)(eitherToSide))
           }
 
-        def step(state: State, a: A): ZIO[R, E1, Step[State, A]] = {
+        def step(state: State, a: A): ZIO[R1, E1, Step[State, A]] = {
           val leftStep: ZIO[R, Nothing, Step[Side[E, self.State, B], A]] =
```

Comment:
` val leftStep: ZIO[R1, ..` ?

Reply:
`leftStep` is actually the step from the current sink, which uses `R`. `rightStep` is from the other sink which uses a narrower `R1`

Reply:
it makes sense, thanks
LGTM